nginx_stage/lib/nginx_stage/user.rb

```diff
@@ -33,11 +33,12 @@ class User
     # @param user [String] the user name defining this object
     # @raise [ArgumentError] if user or primary group doesn't exist on local system
     def initialize(user)
-      @passwd = Etc.getpwnam user.to_s
+      user_is_uid = (user[0].to_i and user.to_i > 0)
```

Comment:
Why check the first character here - `user[0]`?  "123jeff".to_i returns `123` and `jeff123.to_i` returns 0. 

Maybe we need a regex to account for the case when `user` is a proper username that just happens to start with a number. Maybe `user.to_s.chomp.match?(/\A\d+\Z/)`?

Reply:
Usernames are illegal to start with numbers per the POSIX standard (^[a-z][-a-z0-9]*\$").

I was of the impression there were cases that certain strings could return numbers (I am not a Ruby expert), but you are correct, just user.to_i > 0 is shorter. I am away from the desk, but I will change that soon. I didn't want to do a regexp because of the overhead of parsing regexp.

Reply:
I put a new version up that works in various test I did and this will 'fall through' if someone out there has all-number login as their user name. 

I also found a bug in the group conversion. Eg "domain users" is a group name that SSSD can return. So again, I used number id and then map them to an array of strings using Etc.